lib/io/storage/StoragePath.js

```diff
@@ -13,55 +27,260 @@ import ObjectUtils from '@/lib/ObjectUtils';
+    const [arteries, counts] = await Promise.all([
+      ArteryDAO.byStudy(study),
+      CountDAO.byStudy(study),
+    ]);
+    const arteriesByCode = mapBy(arteries, ({ arteryCode }) => arteryCode);
+    const countDirections = counts.map(({ arteryCode }) => {
+      const artery = arteriesByCode.get(arteryCode);
+      return artery.approachDir;
+    });
+    const directions = CardinalDirection.enumValues.filter(
+      direction => countDirections.some(countDirection => countDirection === direction),
+    );
+    return directions
+      .map(({ short }) => `${short}B`)
+      .join('_');
+  }
+
+  /**
+   *
+   * @param {string} id
+   * @returns {string}
+   */
+  static getId(id) {
+    return id.replace('/', '_');
+  }
+
+  /**
+   *
+   * @param {Object} study
+   * @returns {string}
+   */
+  static async getLocationFromStudy(study) {
+    const location = await CentrelineDAO.byFeature(study);
+    const description = StoragePath.sanitizeLocationDescription(location.description);
+    if (location.centrelineType === CentrelineType.SEGMENT) {
+      const partDirections = await StoragePath.getDirectionsFromStudy(study);
+      return `${description}_${partDirections}`;
+    }
+    return description;
+  }
+
+  /**
+   *
+   * @param {Object} data
+   * @returns {string}
+   */
+  static async getLocationsSelection(data) {
+    const { s1, selectionType } = data;
+    const features = CompositeId.decode(s1);
+    const locations = await CentrelineDAO.byFeatures(features);
+    const locationsSelection = { locations, selectionType };
+    const description = getLocationsSelectionDescription(locationsSelection);
+    return StoragePath.sanitizeLocationDescription(description);
+  }
+
+  /**
+   *
+   * @param {Object} options
+   * @returns {string}
+   */
+  static getOptionsHash(options) {
+    if (ObjectUtils.isEmpty(options)) {
+      return '';
+    }
+    const hashBuilder = crypto.createHash('md5');
+    hashBuilder.update(JSON.stringify(options));
+    const hash = hashBuilder.digest('hex').slice(0, 8);
+    return `_${hash}`;
+  }
+
+  /**
+   *
+   * @param {Array<StoragePathResponse>} storagePaths
+   * @returns {string}
+   */
+  static getReportHash(storagePaths) {
+    const hashBuilder = crypto.createHash('md5');
+    const n = storagePaths.length;
+    for (let i = 0; i < n; i++) {
+      const { key: reportKey } = storagePaths[i];
+      hashBuilder.update(reportKey, 'utf8');
+    }
+    return hashBuilder.digest('hex').slice(0, 8);
+  }
+
+  /**
+   *
+   * @param {ReportType} type
+   * @returns {string}
+   */
+  static getReportType(type) {
+    return type.name;
+  }
+
+  /**
+   *
+   * @param {Array<StoragePathResponse>} storagePaths
+   * @returns {string}
+   */
+  static getTotalReports(storagePaths) {
+    return `${storagePaths.length}_TOTAL`;
+  }
+
+  /**
+   *
+   * @param {string} description
+   * @returns {string}
+   */
+  static sanitizeLocationDescription(description) {
+    return description
```

Comment:
this doesn't currently perform any truncation!  we should ensure that it does, and add test cases for this in `StoragePath.spec.js`.